libvast/include/vast/concept/parseable/vast/list.hpp

```diff
@@ -0,0 +1,44 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/aliases.hpp"
+#include "vast/concept/parseable/core.hpp"
+#include "vast/data.hpp"
+
+namespace vast {
+
+struct list_value_parser : parser_base<list_value_parser> {
```

Comment:
list_parser name is already taken. I treat this struct as a sort of implementation detail anyways.

Renaming of list_parser to sequence_parser might also be viable

Reply:
I don't have a strong opinion on this, so choose whatever you prefer.

Reply:
I actually had to remove this. It seems that there is a cyclic dependency between parsing list and parsing data.
So i parse string to record/list with the parsers::data usage.